config/config.go

```diff
@@ -202,3 +195,17 @@ func LoadDefaultConfig(ctx context.Context, optFns ...func(*LoadOptions) error)
 
 	return cfg, nil
 }
+
+func resolveConfigLoaders(options *LoadOptions) []loader {
+	loaders := make([]loader, 2)
+	loaders[0] = loadEnvConfig
+
+	// specification of a profile should cause a load failure if it doesn't exist
+	if os.Getenv(awsProfileEnvVar) != "" || options.SharedConfigProfile != "" {
+		loaders[1] = loadSharedConfig
+	} else {
+		loaders[1] = loadSharedConfigIgnoreNotExist
```

Comment:
am i missing something here? the behavior looks the same: its still using the `loadSharedConfigIgnoreNotExist` which swallows the error (i.e. doesnt return an error message).

Reply:
Unless the check on 204 is `true` though (which means a profile was specified). That's the change here.